Update minimum PHP version (5.2->5.6)

dingo-d opened the pull request:
Update the readme, ruleset and CI for minimum version bump from PHP 5.2 to PHP 5.6.

jrfnl commented on .travis.yml:
```diff
@@ -41,7 +41,7 @@ script:
 
   # Test the ruleset.
-  - vendor/bin/phpcs ./Test/WPTest.php --standard=PHPCompatibilityWP --runtime-set testVersion 5.2
+  - vendor/bin/phpcs ./Test/WPTest.php --standard=PHPCompatibilityWP --runtime-set testVersion 5.6
```
This is not strictly about the version WP supports, but about testing the WP specific _ruleset_, and to be fair, should actually be at `5.0-` to completely test the ruleset.

Until the polyfills which are no longer relevant have been removed from WP, the ruleset won't change and therefore this test shouldn't change.

dingo-d replied:
Oh, so I should revert this to 5.2?

> Until the polyfills which are no longer relevant have been removed from WP, the ruleset won't change and therefore this test shouldn't change.

 Which may not be ever? 😄 I mean, only new versions (5.1+) will require PHP5.6+ if I'm not mistaken. The older WP versions will work on the older PHP versions, right?

jrfnl replied:
> Oh, so I should revert this to 5.2?

Yes please, or alternatively, set it to `5.0-`.

> Which may not be ever?

Very true, don't you just love WP ?

I can imagine that if WP drops those polyfills, we could either do a major release and recommend people who still support older WP versions than the version in which the polyfills were dropped to stay on the 2.x branch, òr alternatively create a secondary ruleset for PHPCompatibilityWP.

Either way, not an issue until WP drops the polyfills.

> I mean, only new versions (5.1+) will require PHP5.6+ if I'm not mistaken. The older WP versions will work on the older PHP versions, right?

Yup, and this ruleset should - for the foreseeable future - still be usable for plugins and themes which would like to continue supporting older WP version.

dingo-d replied:
Thanks for the explanation, I'll set it to `5.0-`.

jrfnl commented on PHPCompatibilityWP/ruleset.xml:
```diff
@@ -3,9 +3,9 @@
 
     <!--
-    The WordPress minimum PHP requirement is PHP 5.2.4.
+    The WordPress minimum PHP requirement is PHP 5.6.
```
I think it would be helpful to mention the "old" requirement as well with the WP version until which it was a requirement. Plugins and themes may still want to support multiple WP versions, so just setting this to `5.6` is misleading.

Suggestion:
```suggestion
    The WordPress minimum PHP requirement was 5.2.4 up to WP 5.1.
    As of WP 5.2, the new minimum PHP requirement is PHP 5.6.
```

jrfnl commented on README.md:
```diff
@@ -18,8 +18,6 @@ This WordPress specific ruleset prevents false positives from the [PHPCompatibil
 
 * [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer).
-    * PHP 5.3+ for use with [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer) 2.3.0+.
-    * PHP 5.4+ for use with [PHP_CodeSniffer](https://github.com/squizlabs/PHP_CodeSniffer) 3.0.2+.
```
These lines should not be removed. The requirements to run PHPCompatibility/PHP_CodeSniffer have nothing to do with the WP minimum version requirements.

jrfnl commented on README.md:
```diff
@@ -61,12 +59,12 @@ By default, you will only receive notifications about deprecated and/or removed
 To get the most out of the PHPCompatibilityWP standard, you should specify a `testVersion` to check against. That will enable the checks for both deprecated/removed PHP features as well as the detection of code using new PHP features.
 
-The minimum PHP requirement of the WordPress project at this time is PHP 5.2.4. If you want to enforce this, either add `--runtime-set testVersion 5.2-` to your command-line command or add `<config name="testVersion" value="5.2-"/>` to your [custom ruleset](https://github.com/PHPCompatibility/PHPCompatibility#using-a-custom-ruleset).
+The minimum PHP requirement of the WordPress project at this time is PHP 5.6. If you want to enforce this, either add `--runtime-set testVersion 5.6-` to your command-line command or add `<config name="testVersion" value="5.6-"/>` to your [custom ruleset](https://github.com/PHPCompatibility/PHPCompatibility#using-a-custom-ruleset).
```
Similar to in the ruleset, you might want to mention the WP version in which the requirements changed here.

jrfnl commented on README.md:
```diff
@@ -86,6 +84,12 @@ All code within the PHPCompatibility organisation is released under the GNU Less
+
+- Ruleset: Changed the comment for minimum PHP version test to 5.6.
+- CI: Changed the ruleset test to pass PHP 5.6.
+- Readme: Updated the minimum required PHP version from 5.2 to 5.6.
```
I appreciate you adding this, but the changelog will be added separately when a new version is released.

If you don't mind, I'll copy this to the draft changelog I already have standing by in a separate branch.

dingo-d commented:
I have changed the things you suggested, I appreciate that :+1: 

Wasn't aware of some of the things you mentioned (test one and the changelog), I have fixed those 🙂
